syntax = "proto3";
package org.kobe.xbot.xdashbackend.utilities;

message ControlPoint {
  double x = 1;
  double y = 2;
}

message BezierCurve {
  repeated ControlPoint controlPoints = 1;
}

message BezierCurves {
  repeated BezierCurve curves = 1;
  bool pathFound = 2;
  optional double metersPerSecond = 3;
  double finalRotationDegrees = 4;
  optional double accelerationMetersPerSecond= 5;
}

message PlanBezierPathRequest {
  ControlPoint start = 1;
  ControlPoint goal = 2;
  double safeRadiusInches = 3;
}